app/src/main/java/com/mybus/activity/MainActivity.java

```diff
@@ -763,26 +766,41 @@ public void onSearchButtonClick() {
     public void onInfoWindowClick(final Marker marker) {
         //Some infoWindow was clicked
         MyBusMarker myBusMarker = null;
-        //Only matters the Start or End location markers
+        //Detect which type of marker is:
         if ((mStartLocationMarker.getMapMarker() != null) && (mStartLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
             myBusMarker = mStartLocationMarker;
         } else if ((mEndLocationMarker.getMapMarker() != null) && (mEndLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
             myBusMarker = mEndLocationMarker;
-        }
+        } else if (mUserLocationMarker.getMapMarker() != null && (mUserLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
```

Comment:
We can refactor this into a method, passing the marker we want to check and the MyBusMarker in order to avoid the cheking != null and the whole "someMyBusMarker.getMapMarker().getId()"

`private boolean isMarkerEquals(Marker marker, MyBusMarker myBusMarker) {
        return myBusMarker.getMapMarker() != null && myBusMarker.getMapMarker().getId().equals(marker);
    }`